### ISSUE TITLE
Shader Management Console Python Script Fails On Name Class

### ISSUE THREAD

**Issue by santorac** [o3de/o3de#13137]
**Describe the bug**
The Shader Management Console script GenerateShaderVariantListForMaterials.py reports a bunch of python binding warnings and fails to produce the expected shader variant data. It is not possible to generate .shadervariantlist files, the workflow is **blocked**.

**Steps to reproduce**
1. Run Shader Management Console
2. In the asset browser, find BasePBR_ForwardPass.shader
3. Right-click > Python Scripts > GenerateShaderVariantListForMaterials.py

**Expected behavior**
After a moment, a new document should be opened with a table of shader option values.

**Actual behavior**
A new document is opened with table that has no values. The log has a bunch of messages like this:
```
==================================================================
python: Trace::Warning
 D:/o3de/Gems/EditorPythonBindings/Code/Source/PythonMarshalComponent.cpp(669): 'bool __cdecl EditorPythonBindings::Container::LoadPythonToPairElement(struct _object *,enum AZ::BehaviorParameter::Traits,const struct AZ::SerializeContext::ClassElement *,class AZ::SerializeContext::IDataContainer *,unsigned __int64,class AZ::SerializeContext *,void *)'
python: Could not convert to pair element type value1 for the pair<>; failed to marshal Python input <class 'str'>
python: ==================================================================
python: 
==================================================================
python: Trace::Warning
 D:/o3de/Gems/EditorPythonBindings/Code/Source/PythonMarshalComponent.cpp(669): 'bool __cdecl EditorPythonBindings::Container::LoadPythonToPairElement(struct _object *,enum AZ::BehaviorParameter::Traits,const struct AZ::SerializeContext::ClassElement *,class AZ::SerializeContext::IDataContainer *,unsigned __int64,class AZ::SerializeContext *,void *)'
python: Could not convert to pair element type value2 for the pair<>; failed to marshal Python input <class 'str'>
python: ==================================================================
python: 
==================================================================
python: Trace::Warning
 D:/o3de/Gems/EditorPythonBindings/Code/Source/PythonMarshalComponent.cpp(669): 'bool __cdecl EditorPythonBindings::Container::LoadPythonToPairElement(struct _object *,enum AZ::BehaviorParameter::Traits,const struct AZ::SerializeContext::ClassElement *,class AZ::SerializeContext::IDataContainer *,unsigned __int64,class AZ::SerializeContext *,void *)'
python: Could not convert to pair element type value1 for the pair<>; failed to marshal Python input <class 'str'>
python: ==================================================================
```

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
Seen on multiple computers.

**Additional context**
It appears the "pair" that it is trying to handle is AZStd::pair<AZ::Name, AZ::Name>

**Comment by santorac** [comment]
Note that even though this issue appears in a graphics tool, I think this is a @o3de/sig-core issue.

**Comment by etherealchain** [comment]
I am able to fix this in #12898, by passing in `AZStd::pair<AZ::Name, AZ::Name>` instead of `AZStd::pair<AZStd::string, AZStd::string>` on line 111 of GenerateShaderVariantListForMaterials.py
Note this will create another warning
```
python: Trace::Warning
 D:/o3de/Gems/EditorPythonBindings/Code/Source/PythonProxyObject.cpp(462): 'class pybind11::object __cdecl EditorPythonBindings::PythonProxyObject::GetWrappedObjectRepr(void)'
python: The type (Name) does not implement the __repr__ method.
python: ==================================================================
```

But the list can be generated correctly
